web/src/engine/websites/CopyManga.ts

```diff
@@ -0,0 +1,161 @@
+        return new RegExp(`^${this.URI.origin}/comic/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = new URL(url).pathname.split('/').pop();
+        const request = this.createApiRequest(`api/v3/comic2/${id}`);
+        const data = await FetchJSON<APIResponse<APISingleComic>>(request);
+        return new Manga(this, provider, data.results.comic.path_word, data.results.comic.name.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        for (let page = 0, run = true; run; page++) {//its offset so page start at 0
+            const mangas = await this.getMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    private async getMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        try {
+            const request = this.createApiRequest(`/api/v3/comics?ordering=-datetime_updated&limit=50&offset=${page * 50}`); //50 is the max
+            const data = await FetchJSON<APIResponse<APIResultList<APIComic>>>(request);
+            return data.results.list.map(item => {
+                return new Manga(this, provider, item.path_word, item.name.trim());
+            });
+        } catch (error) {
+            return [];
+        }
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const chapterList = [];
+        for (let page = 0, run = true; run; page++) {
+            const chapters = await this.getChaptersFromPage(manga, page);
+            chapters.length > 0 ? chapterList.push(...chapters) : run = false;
+        }
+        return chapterList;
+    }
+
+    private async getChaptersFromPage(manga: Manga, page: number): Promise<Chapter[]> {
+        try {
+            const request = this.createApiRequest(`/api/v3/comic/${manga.Identifier}/group/default/chapters?limit=500&offset=${page * 500}`);
+            const data = await FetchJSON<APIResponse<APIResultList<APIChapter>>>(request);
+            return data.results.list.map(item => {
+                return new Chapter(this, manga, item.uuid, item.name.trim());
+            });
+        } catch (error) {
+            return [];
+        }
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const request = this.createApiRequest(`/api/v3/comic/${chapter.Parent.Identifier}/chapter2/${chapter.Identifier}?platform=3`);
+        const data = await FetchJSON<APIResponse<APIPages>>(request);
+        const imageUrls = data.results.chapter.contents.map(item => item.url);
+        const imageOrder = data.results.chapter.words;
+        const orderedPages : string[] = [];
+        imageOrder.forEach((order, index) => {
+            orderedPages[order] = imageUrls[index];
```

Comment:
The variable `order` is of type string and not number, i would expect that it does not work correctly if this is not a number. Shouldn't it be sorted in some way to work?

Reply:
linter and editor tell me `order ` is a number  but ill double check.

Reply:
Its a number there is no problem. `data.results.chapter.words` is an array of numbers after all.
There is no way to properly order image just by using their url.